File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Python/s196249385_commented.py
import sys

def main():
    n, k = map(int, sys.stdin.readline().split())
    a = list(map(int, sys.stdin.readline().split()))
    
    cumsum = [0] * (n + 1)
    for i in range(n):
        cumsum[i + 1] = cumsum[i] + a[i]
    
    b = []
    max_val = 0
    for i in range(n):
        for j in range(i, n):
            s = cumsum[j + 1] - cumsum[i]
            b.append(s)
            if s > max_val:
                max_val = s
    
    if max_val == 0:
        mask = 0
    else:
        mask = 2 ** ((max_val).bit_length() - 1)
    
    current_mask = mask
    while current_mask > 0:
        tmp = []
        for e in b:
            if (e & current_mask) != 0:
                tmp.append(e)
        if len(tmp) >= k:
            b = tmp
        current_mask >>= 1
    
    if len(b) < k:
        print(0)
    else:
        ans = -1
        for e in b:
            ans &= e
        print(ans)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s196249385_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"math/bits"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	var n, k int
	if sc.Scan() {
		n, _ = strconv.Atoi(sc.Text())
	} else {
		fmt.Fprintln(os.Stderr, "Error reading n")
		os.Exit(1)
	}
	if sc.Scan() {
		k, _ = strconv.Atoi(sc.Text())
	} else {
		fmt.Fprintln(os.Stderr, "Error reading k")
		os.Exit(1)
	}

	var a []int
	a = make([]int, n)
	for i := 0; i < n; i++ {
		if sc.Scan() {
			num, _ := strconv.Atoi(sc.Text())
			a[i] = num
		} else {
			fmt.Fprintln(os.Stderr, "Error reading array elements")
			os.Exit(1)
		}
	}

	var cumsum []int64
	cumsum = make([]int64, n+1)
	for i := 0; i < n; i++ {
		cumsum[i+1] = cumsum[i] + int64(a[i])
	}

	var b []int64
	var max int64 = 0
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			sum := cumsum[j+1] - cumsum[i]
			b = append(b, sum)
			if sum > max {
				max = sum
			}
		}
	}

	var mask int64
	if max == 0 {
		mask = 0
	} else {
		mask = 1 << (bits.Len64(max) - 1)
	}

	for mask > 0 {
		var tmp []int64
		for _, e := range b {
			if (e & mask) != 0 {
				tmp = append(tmp, e)
			}
		}
		if len(tmp) >= k {
			b = tmp
		}
		mask >>= 1
	}

	if len(b) < k {
		fmt.Println(0)
	} else {
		ans := int64(-1)
		for _, e := range b {
			ans &= e
		}
		fmt.Println(ans)
	}
}

// End of Code
